components/provider_ppucd/src/structs.rs

```diff
@@ -0,0 +1,130 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+use icu_uniset::UnicodeSet;
+use std::convert::TryInto;
+
+//
+// resource key structs - the structs used directly by users of data provider
+//
+
+pub mod key {
+    use icu_provider::resource_key;
+    use icu_provider::ResourceKey;
+
+    pub const AHEX_V1: ResourceKey = resource_key!(uniset, "AHex", 1);
```

Comment:
Where did you get these string names?  They don't look consistent; some are all lowercase, some are all caps, and some are mixed.

Reply:
I got them from the [ppucd.txt file](https://raw.githubusercontent.com/unicode-org/icu/master/icu4c/source/data/unidata/ppucd.txt) itself - for every line that begins `property;binary;...`, I took the first listed alias name as the canonical name for that binary property name, and kept the upper-/lower-casing as-is for the associated name string.

Reply:
Make sure there is a follow-up issue so we can discuss this with Markus.